### ISSUE TITLE
Use dim instead of gray in formatters for compatibility with color chaining

### ISSUE THREAD

**Issue by kriansa** [stylelint/stylelint#1996]
When using iterm2 solarized theme, string formatter is displaying line number so light we can barely see it.

Currently, it uses `chalk.gray`, but I think that `chalk.dim.white` would be better. What do you think?

![screenshot](https://i.imgsafe.org/4455877f84.png)

**Comment by jeddy3** [comment]
Is this not a bug with the iterm2 solarized theme as `gray` should be gray and not black, right?

Having said that, can you go into a bit more detail about why you think `dim.white` would be better than `gray` e.g. is there a resource out there that lays out the compatibility of each?

**Comment by kriansa** [comment]
In fact, that seems to be a solarized issue, as `gray` is not a compatible color.

https://github.com/altercation/solarized/issues/220

I just fixed my issue by installing a forked version of this theme ([base16-solarized](https://github.com/chriskempson/base16-iterm2/blob/master/base16-solarized.dark.256.itermcolors))

However, I'd like to add more to this discussion.

I think that using `dim.white` is not ideal. Instead, I suggest using just `dim` because it gives a similar look than it is now and also allow external color chaining, for instance, if you call:

```js
chalk.yellow(formater(output))
```

The final color would be `dim.yellow`. That is not possible, however, if you explicitely set the color upfront. By the way, this is what [stylelint-webpack-plugin](https://github.com/vieron/stylelint-webpack-plugin/blob/master/lib/run-compilation.js#L30) is doing.

ESLint is already using `dim`. (https://github.com/eslint/eslint/blob/master/lib/formatters/stylish.js#L76)

**Comment by jeddy3** [comment]
Thanks for adding more to the discussion.

>  I suggest using just dim because ... allow external color chaining

That's a compelling reason to change to `dim` from `gray`.

@stylelint/core Any objections to making this change?

**Comment by davidtheclark** [comment]
No objection.

**Comment by kriansa** [comment]
Would you like me to submit a PR for that?

**Comment by jeddy3** [comment]
If you wouldn't mind. 

All the issues labeled with "status: help wanted" are up for grabs :)
